File: src/main/java/com/cag/cagbackendapi/entities/PastPerformanceEntity.java
package com.cag.cagbackendapi.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.GenericGenerator;
import org.hibernate.annotations.Type;

import javax.persistence.*;
import java.util.UUID;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "past_performance")
public class PastPerformanceEntity {
    @Id
    @Type(type = "pg-uuid")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(
            name = "UUID",
            strategy = "org.hibernate.id.UUIDGenerator"
    )
    private UUID past_performance_id;
    private String show_title;
    private String role;
    private String theater_or_location;
    private String show_url;
    private String director;
    private String musical_director;
    private String theater_group;

    @OneToOne
    @JoinColumn(foreignKey = @ForeignKey(name = "fk_profile"))
    private ProfileEntity profileEntity;
}

File: src/main/java/com/cag/cagbackendapi/entities/PastPerformanceEntity.kt
package com.cag.cagbackendapi.entities

import org.hibernate.annotations.GenericGenerator
import org.hibernate.annotations.Type
import java.util.*
import javax.persistence.*

@Entity
@Table(name = "past_performance")
data class PastPerformanceEntity (
    @Id
    @Type(type = "pg-uuid")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(
            name = "UUID",
            strategy = "org.hibernate.id.UUIDGenerator"
    )
    var past_performance_id: UUID?,
    var show_title: String?,
    var role: String?,
    var theater_or_location: String?,
    var show_url: String?,
    var director: String?,
    var musical_director: String?,
    var theater_group: String?,

    @OneToOne
    @JoinColumn(foreignKey =ForeignKey(name = "fk_profile"))
    var profileEntity: ProfileEntity?
)
